packages/playwright-test/src/matchers/golden.ts

```diff
@@ -129,8 +129,8 @@ export function compare(
       return { pass: true, message };
     }
     if (updateSnapshots === 'missing') {
-      testInfo._appendErrorMessage(message);
-      return { pass: true, message };
+      testInfo._failWithError(serializeError(new Error(message)), false /* isHardError */);
```

Comment:
Does this get a nice stack trace?

Reply:
It does, thanks to a new stack filter that filters out expect stack frames.